runtime/reconcilers/metrics_view.go

```diff
@@ -152,6 +153,36 @@ func (r *MetricsViewReconciler) validate(ctx context.Context, mv *runtimev1.Metr
 	return errors.Join(errs...)
 }
 
+func validateDimension(
+	ctx context.Context,
+	olap drivers.OLAPStore,
+	t *drivers.Table,
+	d *runtimev1.MetricsViewSpec_DimensionV2,
+	fields map[string]*runtimev1.StructType_Field,
+) error {
+	if d.Expression == "" {
+		if d.Column == "" {
+			return nil
+		}
+		if _, isColumn := fields[d.Column]; !isColumn {
+			return fmt.Errorf("column not found: %s", d.Column)
```

Comment:
Better error message – should include dimension name, something like `failed to validate dimension %q: column %q not found in table`